docs/source/tutorials/writing-a-plugin-for-hermes.md

```diff
@@ -0,0 +1,156 @@
+HermesSettings are the base class for command specific settings.
+It uses pydantic settings to specify and validate the parameters.
+The user can either set the parameters in the hermes.toml or overwrite them in the command line.
+To overwrite the configuration, you use the -O operator with the dotted parameter name and the value.
+
+## Set Up Plugin
+To write a new plugin, it is important to follow the given structure.
+This means your plugins source code has a pydantic class with Settings and the plugin class which inherits from one base class.
+For our specific case, we want to write a git harvest plugin.
+Our class Structure should look like this:
+
+
+```{code-block} python
+from hermes.commands.harvest.base import HermesHarvestPlugin
+from pydantic import BaseModel
+
+
+class GitHarvestSettings(BaseModel):
+    from_branch: str = 'main'
+
+
+class GitHarvestPlugin(HermesHarvestPlugin):
+    settings_class = GitHarvestSettings
+
+    def __call__(self, command):
+        print("Hello World!")
+
+        return {}, {}
+```
+ 
+The Code uses the HermesHarvestPlugin as base class and pydantics Basemodel for the Settings. In the GitHarvestSettings you
+can see that one setting is made. The Parameter from_branch is specific for this plugin and can be reached through self.settings.harvest.git.git_branch as long as our plugin will be named git.
+In the hermes.toml this would be achieved by [harvest.{plugin_name}].
+The GitHarvestSettings are assigned to the GitHarvestPlugin. In the plugin you need to overwrite the __ call __ method.
+For now a simple Hello World will do. The method return two dictionaries. These will later depict the harvested data in codemeta (json-ld) and information for generating hermes metadata.
+That is the basic structure for the plugins source code.
+
+To integrate this code, you have to register it as a plugin in the pyproject.toml. To learn more about the pyproject.toml check https://python-poetry.org/docs/pyproject/.
+We will just look at the important places for this plugin. There are two ways to integrate this plugin. First we will show how to use the plugin environment as the running base with HERMES as a dependency.
+Then we say how to integrate this plugin in HERMES itself.
+
+### Include HERMES as Dependency
+This is probably the more common way, where you can see HERMES as a framework.
+The idea is that your project is the main part. You create the pyproject.toml as usual.
+In the dependencies block you need to include hermes. Then you just have to declare your plugin.
+The HERMES software will look for plugins and install these.
+In the code below you can see the parts of the pyproject.toml that are important.
+```{code-block} toml
+...
+[tool.poetry.dependencies]
+python = "^3.10"
+hermes = "^0.8.0"
+...
+...
+[tool.poetry.plugins."hermes.harvest"]
+git = "hermes_git.harvest:GitHarvestPlugin"
+...
+```
+As you can see the plugin class from hermes_git is declared as git for hermes.harvest.
```

Comment:
```suggestion
As you can see the plugin class from `hermes_git` is declared as `git` for the `hermes.harvest` entrypoint.
```

Reply:
Maybe we should change that to hermes_plugin_git?